src/net_processing.cpp

```diff
@@ -2892,17 +2892,15 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         // fProvidesHeaderAndIDs is used to "lock in" version of compact blocks we send (fWantsCmpctWitness)
         if (!State(pfrom.GetId())->fProvidesHeaderAndIDs) {
             State(pfrom.GetId())->fProvidesHeaderAndIDs = true;
-            State(pfrom.GetId())->fWantsCmpctWitness = nCMPCTBLOCKVersion == 2;
+            State(pfrom.GetId())->fWantsCmpctWitness = true;
```

Comment:
nit: I don't see why 3223906fb7a50d089308c49c9f63e88982c515c2 is split up. In the line before you are ensuring that the version is only 2, but you leave this line untouched, which makes it confusing to review the previous commit.

Might be best to squash.

Reply:
I tried to split this PR into small commits after this comment: https://github.com/bitcoin/bitcoin/pull/20799#issuecomment-752197637. I'm not sure if there's a way to structure these changes that is going to satisfy everyone, so I'd prefer to leave this as is.